### DISCUSSION TITLE
Bug with Concatenated tables

### DISCUSSION THREAD

**Discussion by friosavila** [quarto-dev/quarto-cli#10000] [Q&A] [answered]
Hi, 
I think i found a bug when trying to created multiple tables and render them into latex.

Here the example
```
---
title: "Untitled"
format: 
    pdf:
        header-includes: | 
            \usepackage{lipsum}
---

## Case 1

\lipsum[1-2]

:::{#tbl-tm1 tbl-pos="H"}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Nice Table: Right where it belongs
:::
\lipsum[1-2]

## Case 2

\lipsum[1-2]

:::{#tbl-tm2 tbl-pos="H" layout-ncol="2"}

:::{#tbl-tm2a}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table1: 
:::

:::{#tbl-tm2b}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table2: 
:::

Not so nice: Where Latex Wants it
:::

\lipsum[1-2]

```

Interestingly, this doesn't happen if i do not use layout-ncol="2".

For Reference, I'm using Quarto 1.4.554

and here the rendered Document
[Untitled-3.pdf](https://github.com/user-attachments/files/15826311/Untitled-3.pdf)

**Comment by cderv** [comment]
It seems to be related to `tbl-pos` handling. 

````markdown
---
title: "Untitled"
format: pdf
keep-tex: true
---

## Case 1

{{< lipsum 1 >}}

:::{#tbl-tm tbl-pos="H"}

:::{#tbl-tma}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table1: 
:::

:::{#tbl-tmb}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table2: 
:::

This table as `tbl-pos`
:::

{{< lipsum 1-2 >}}

:::{#tbl-tm2 tbl-pos="H"}

::: {layout-ncol=2}

:::{#tbl-tm2a}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table1: 
:::

:::{#tbl-tm2b}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table2: 
:::
:::

This table also as `tbl-pos` an a layout
:::

{{< lipsum 1-2 >}}

:::{#tbl-tm3 tbl-pos="H" layout-ncol=2}

:::{#tbl-tm3a}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table1: 
:::

:::{#tbl-tm3b}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table2: 
:::

This table as lost `tbl-pos`
:::
````

- First table have `\begin{table}[H]`
- Second table also have `\begin{table}[H]`
- But latest one does not : `\begin{table}`

It seems the latest syntax should be working... 🤔 

@cscheid do you confirm it should and this is an issue ?

**Reply by cderv** [reply, marked as answer]
In fact the issue is because we are using `fig-pos` here, and not `tbl-pos`
https://github.com/quarto-dev/quarto-cli/blob/50b4cf15a0be98a439773ffe923e1e284301879b/src/resources/filters/layout/latex.lua#L79-L87

So this works 
````markdown
---
title: "Untitled"
format: pdf
keep-tex: true
---

## Case 1

{{< lipsum 1-2 >}}

:::{#tbl-tm3 fig-pos="H" layout-ncol=2}

:::{#tbl-tm3a}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table1: 
:::

:::{#tbl-tm3b}
| A | B | C |
|---|---|---|
| 1 | 2 | 3 |

Table2: 
:::

This table as lost `tbl-pos`
:::

{{< lipsum 1-2 >}}
 
````

I think we should compute based on the prefix of the float ref 
Like in here: 
https://github.com/quarto-dev/quarto-cli/blob/50b4cf15a0be98a439773ffe923e1e284301879b/src/resources/filters/layout/latex.lua#L513-L527

I'll move to an issue

**Reply by mcanouil** [reply]
- https://github.com/quarto-dev/quarto-cli/issues/10001
